### PR TITLE
Test lakeFS with direct-access "thick" Spark client

### LINKED ISSUE #1921: Test Spark data client during Nessie Spark test

### PR DESCRIPTION
**Pull request by arielshaqed**
Fixes #1921.

### CODE REVIEW

**Comment by arielshaqed** [comment]
Note that test still doesn't _pass_.  Pulling it allows people to work on the many ways it can fail, as well as keep sync'ed on progress and separate fixes into multiple PRs.

File: test/spark/run-test.sh [hunk 1]
```diff
@@ -3,18 +3,25 @@
...
 #   STORAGE_NAMESPACE - storage namespace used for repository creation
+#   USE_DIRECT_ACCESS - if set (to *anything*), use the direct thick Spark client
 # 
 # NOTE that this script should be run from the root project in order for docker compose to volume mount the project
 
 STORAGE_NAMESPACE=${STORAGE_NAMESPACE:-local://}
 
-docker-compose exec -T lakefs /app/wait-for localhost:8000
+export input="lakefs://${REPO}/main/sonnets.txt"
+export output="lakefs://${REPO}/main/sonnets-wordcount.csv"
```
Comment (nopcoder) [inline test/spark/run-test.sh:15]: why do we need these as exported env variables - can we just pass the values without defining env variable for each data we pass?

File: test/spark/docker-compose.yaml [hunk 2]
```diff
@@ -61,18 +63,40 @@ services:
...
     profiles: ["command"]
     volumes:
       - ./:/local
+    env_file:
+      - tester.env
+    environment:
+      - input
+      - output
+      - s3input
+      - s3output
```
Comment (nopcoder) [inline test/spark/docker-compose.yaml:89]: if possible I prefer not to pass these and env - it should of the command we execute part of the command we run.

File: .github/workflows/nessie.yaml [hunk 3]
```diff
@@ -144,13 +144,43 @@ jobs:
...
+          JARS: clients/hadoopfs/
+          STORAGE_NAMESPACE: s3://nessie-system-testing/${{ github.run_number }}-spark-client
+          AWS_ACCESS_KEY_ID: ${{ secrets.NESSIE_AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.NESSIE_AWS_SECRET_ACCESS_KEY }}
+          USE_DIRECT_ACCESS: "true"
+          REPO: thick-client-test
+        working-directory: test/spark
+        run: ./run-test.sh      # TODO(ariels): import
+                                # ../../clients/hadoopfs/target/client.jar into the
+                                # containers!
```
Comment (nopcoder) [inline .github/workflows/nessie.yaml:181]: why do we need to import - spark-submit doesn't copy the jar for us?

**Review by nopcoder — CHANGES_REQUESTED** [review]

**Comment by nopcoder** [comment]
A question and request to remove the extra environment variables (input/output) from the game.

**Review by nopcoder — APPROVED** [review]

**Comment by arielshaqed** [comment]
Thanks!

**Pull request merged**